src/main/java/stellarium/DrawSky.java

```diff
@@ -321,11 +326,12 @@ else if (mc.theWorld.provider.isSurfaceWorld())
 					double longdd=(double)longcd/(double)longn;
 					double latdd=1.0-(double)(latc+1)/(double)latn;
 
-					tessellator1.setColorRGBA_F(1.0f, 1.0f, 1.0f, (f4*moonilum[longc][latc]-0.15f*bglight)*2.0f);
+					float lightlevel = (0.875f*(bglight/2.1333334f));
+					tessellator1.setColorRGBA_F(1.0f - lightlevel, 1.0f - lightlevel, 1.0f - lightlevel, ((f4*moonilum[longc][latc]-0.015f*bglight)*2.0f));
```

Comment:
It goes back to what I explained about Alpha not always being between 0 & 1. Alpha can be below 0, or above 1. In the case of the Moon, the alpha is extremely high, that's why it doesn't fade out during the day.

lightlevel is based on how much light there is from 0 to 1. That's why I use it instead.